geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/ConverterInputFormat.scala

```diff
@@ -86,13 +96,25 @@ class ConverterRecordReader extends FileStreamRecordReader with LazyLogging {
       case None    => raw
     }
 
+    import scala.collection.JavaConversions._
+    val featureReader = if (retypedSpec != null) {
+      val retypedSft = SimpleFeatureTypes.createType(sft.getTypeName, retypedSpec)
+      val reader = new DelegateSimpleFeatureReader(sft, new DelegateSimpleFeatureIterator(iter))
+      new ReTypeFeatureReader(reader, retypedSft)
+    } else {
+      new DelegateSimpleFeatureReader(sft, new DelegateSimpleFeatureIterator(iter))
+    }
+
     logger.info(s"Initialized record reader on split ${filePath.toString} with " +
         s"type name ${sft.getTypeName} and convert conf $confStr")
 
     new Iterator[SimpleFeature] with Closeable {
-      override def hasNext: Boolean = iter.hasNext
-      override def next(): SimpleFeature = iter.next
-      override def close(): Unit = converter.close()
+      override def hasNext: Boolean = featureReader.hasNext
+      override def next(): SimpleFeature = featureReader.next
+      override def close(): Unit = {
+        featureReader.close()
```

Comment:
I prefer to encase these with CloseWithLogging() or CloseQuietly() from geomesa utils